app/views/teachers/classes/CourseProgressDotView.vue

```diff
@@ -0,0 +1,266 @@
+        border-right-color: transparent !important;
+        border-bottom-color: transparent !important;
+        top: -6px;
+        left: calc(50% - 10px);
+
+        margin-top: 0;
+        margin-bottom: 0;
+    }
+</style>
+
+<template>
+    <li>
+        <v-popover
+                popover-base-class="course-dot-progress-tooltip"
+                popover-arrow-class="arrow"
+                trigger="hover"
+                placement="top"
+                :open-group="`${classroom._id}${course._id}`"
+        >
+            <div :class="[ 'progress-dot', { started: courseStarted, completed: courseCompleted } ]">
+                {{ courseAcronym }}
+            </div>
+
+            <template slot="popover">
+                <span v-if="levelSessionsLoading">
+                    {{ $t('common.loading') }}
+                </span>
+                <div v-else>
+                    {{ courseStats.studentsCompletingAllLevels }} / {{ classroom.members.length }}
+                    {{ $t('courses.students')}}
+                    <br />
+                    {{ percentCompleted }}%
+                    {{ $t('teacher.completed') }}
+                </div>
+            </template>
+        </v-popover>
+    </li>
+</template>
+
+<script>
+  import { mapState } from 'vuex'
+
+  export default {
+    props: {
+      course: Object,
+      classroom: Object,
+    },
+
+    computed: Object.assign({},
+      // TODO this could be loading (top level component prevents this now but may not in future).  Handle loading state here
+      mapState('courses', {
+        courses: function (state) {
+          return state.byId
+        }
+      }),
+
+      mapState('courseInstances', {
+        // TODO this could be loading (top level component prevents this now but may not in future).  Handle loading state here
+        courseInstancesLoading (state) {
+          return state.loading.byTeacher[this.$props.classroom.ownerID]
+        },
+
+        courseInstance (state) {
+          const instances = state.courseInstancesByTeacher[this.$props.classroom.ownerID] || []
+
+          return instances
+            .find(i => i.courseID === this.$props.course._id && i.classroomID === this.$props.classroom._id)
+        }
+      }),
+
+      mapState('levelSessions', {
+        // TODO this could be loading (top level component prevents this now but may not in future).  Handle loading state here
+        levelSessionsLoading (state) {
+          return state.loading.sessionsByClassroom[this.$props.classroom._id]
+        },
+
+        levelCompletionsByUser (state) {
+          const levelSessionState = state.levelSessionsByClassroom[this.$props.classroom._id]
+
+          if (levelSessionState) {
+            return levelSessionState.levelCompletionsByUser
+          }
+
+          return {}
+        }
+      }),
+
+      {
+        // TODO course acronym could be controlled and sent from the backend
+        courseAcronym: function () {
```

Comment:
We have course acronyms defined in the `utils` here: https://github.com/codecombat/codecombat/blob/master/app/core/utils.coffee#L136
Can use that here.

Reply:
This is great, so to make sure I understand, that is a mapping from course ID to acronym meaning that we can simply reference this map?

Reply:
Yeah I believe so

Reply:
Updated